File: python/examples/parse_serialize.py
"""Parse Turtle and serialize N-Triples."""

from oxiland import Model, load, serialize

model = Model()
n = load(
    model,
    '<https://example.com/alice> <https://example.com/name> "Alice" .',
    "turtle",
)
assert n == 1
text = serialize(model, "ntriples")
assert "Alice" in text
print("parse_serialize ok")

File: examples/parse_serialize.rs
//! Parse Turtle and serialize N-Triples with the Oxiland I/O facade.

use oxiland::Model;
use oxiland::io::{Parser, Serializer, Syntax};

fn main() -> oxiland::Result<()> {
    let model = Model::new()?;
    Parser::for_syntax(Syntax::Turtle)
        .base_iri("https://example.com/")?
        .load_collecting(&model, b"<alice> <name> \"Alice\" .".as_slice())?;

    let ntriples = Serializer::for_syntax(Syntax::NTriples).serialize_model_to_string(&model)?;
    println!("{ntriples}");
    Ok(())
}
